tests/pages/firefox/developer.py

```diff
@@ -0,0 +1,50 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as expected
+from selenium.webdriver.support.ui import WebDriverWait as Wait
+
+from .base import FirefoxBasePage
+from ..page import PageRegion
+
+
+class DeveloperPage(FirefoxBasePage):
+
+    _url = '{base_url}/{locale}/firefox/developer'
+    _primary_download_locator = (By.CSS_SELECTOR, '.intro .download-button')
+    _secondary_download_locator = (By.CSS_SELECTOR, '.dev-footer-download .download-button')
+
+    @property
+    def is_primary_download_button_displayed(self):
+        return self.selenium.find_element(*self._primary_download_locator).is_displayed()
+
+    @property
+    def is_secondary_download_button_displayed(self):
+        return self.selenium.find_element(*self._secondary_download_locator).is_displayed()
+
+    @property
+    def developer_videos(self):
+        return [DeveloperVideoModal(self.selenium)]
+
+
+class DeveloperVideoModal(PageRegion):
```

Comment:
I think this sounds like a good abstraction, makes sense to have a generic `Modal` PageRegion since this is the same JS plugin used throughout the site. They don't always play videos, but do use common elements in the DOM.